### DISCUSSION TITLE
Hide Navbar and footer for the Login and Registration page

### DISCUSSION THREAD

**Discussion by pranavp10** [vercel/next.js#16865] [Help] [answered]
I am using _app.js to so that i can diplay navbar and footer to all the pages but i want to disable the for the login page and registration page 
 
_app.js
`import { useRouter } from 'next/router';`
`function Myapp({Component, pageProps}){`
`const router = useRouter();`
  `const { asPath } = router;`
  `const noNav =  ['/login', '/Registration'];`
  `return(`
   `       {noNav.includes(asPath) ? null : <Nav />}`
  `          <Component {...pageProps} />`
 `         {noNav.includes(asPath) ? null : <Footer />}`
`);`
`}`

The above example does does hide the nav and footer but 
the issure is with if the path is `localhost/login?sometext` its displaying the navbar 
Help willappreciated 🙏

**Comment by jensmeindertsma** [comment, marked as answer]
You should use `router.path` instead of `router.asPath` for this use case. Can you try this?

**Reply by pranavp10** [reply]
nop its not working :(
thanks for helping

**Reply by creativesuraj** [reply]
I think @jensmeindertsma means `router.pathname`. asPath contains the actual path and your noNav does not contain the actual path, it contains the pathnames so `router.pathname` works here.

**Reply by pranavp10** [reply]
thanks suraj :) for elaborating it.

**Reply by best2025j** [reply]
thanks man your code did work

**Reply by BigOk3y** [reply]
Hi @best2025j, Can I see the _app.js of the correction?

**Reply by BigOk3y** [reply]
I've gotten it
no need anymore
Thanks anyways

**Reply by best2025j** [reply]
alright

**Reply by bashrian** [reply]
it worked thanks

**Reply by DeepakGathwal** [reply]
× You have a Server Component that imports next/router. Use next/navigation instead.\nLearn more: https://nextjs.org/docs/app/api-reference/functions/use-router
 
 nextJs 14

**Comment by best2025j** [comment]
alright

**Comment by vamshig9** [comment]
Nice it's Working

**Comment by best2025j** [comment]
Great, thanks for letting me know.
